node/node.go

```diff
@@ -578,7 +581,9 @@ func (node *AlgorandFullNode) GetPendingTransaction(txID transactions.Txid) (res
 // Status returns a StatusReport structure reporting our status as Active and with our ledger's LastRound
 func (node *AlgorandFullNode) Status() (s StatusReport, err error) {
 	node.mu.Lock()
```

Comment:
It would be sufficient to take the new lock during the setup of these two output variables, i.e.
```
node.syncMu.Lock()
s.LastRoundTimestamp = node.lastRoundTimestamp
s.HasSyncedSinceStartup = node.hasSyncedSinceStartup
node.mu.Unlock()

node.mu.Lock()
defer node.syncMu.Unlock()
...
```